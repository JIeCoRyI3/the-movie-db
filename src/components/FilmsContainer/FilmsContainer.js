import React from 'react';
import PropTypes from 'prop-types';
import FilmCard from '../FilmCardComponent/';

const FilmsContainer = (props) =>
    props.films.map((film, index) => (
        <FilmCard
            key={index}
            poster_path={film.poster_path}
            genres={film.genre_ids}
            release_date={film.release_date}
            title={film.title}
        />
    ));

FilmsContainer.propTypes = {
    films: PropTypes.arrayOf(
        PropTypes.shape({
            poster_path: PropTypes.string,
            genres_ids: PropTypes.arrayOf(PropTypes.number),
            release_date: PropTypes.string,
            title: PropTypes.string.isRequired,
        })
    ),
};

export { FilmsContainer };
