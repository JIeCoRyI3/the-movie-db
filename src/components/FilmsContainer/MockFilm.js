import React from 'react';

function MockFilm() {
    return (
        <div
            style={{
                width: '200px',
                height: '300px',
                border: '2px solid red',
                borderRadius: '5px',
            }}
        >
            1
        </div>
    );
}

export default MockFilm;
