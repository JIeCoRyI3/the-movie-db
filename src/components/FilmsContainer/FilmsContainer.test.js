import { shallow } from 'enzyme';
import { FilmsContainer } from './FilmsContainer';
import React from 'react';

describe('<FilmCard>', () => {
    let wrapper;

    const props = {
        films: ['FilmCard', 'FilmCard', 'FilmCard'],
    };

    beforeEach(() => {
        wrapper = shallow(<FilmsContainer {...props} />);
    });

    it('should render <FilmsContainer/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
