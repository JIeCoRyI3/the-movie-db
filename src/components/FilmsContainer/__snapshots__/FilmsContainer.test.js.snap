// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmCard> should render <FilmsContainer/> correctly 1`] = `undefined`;
